haskell/toolchain.bzl

```diff
@@ -56,6 +70,41 @@ def _run_ghc(hs, cc, inputs, outputs, mnemonic, arguments, env, params_file = No
     hs.actions.write(compile_flags_file, args)
     hs.actions.write(extra_args_file, arguments)
 
+    if abi_file != None:
+        # We declare the file containing informations about the call to GHC with --show-iface.
+        show_iface_file = hs.actions.declare_file("show_iface_%s_%s" % (hs.name, extra_name))
+        show_iface_args = hs.actions.args()
+        show_iface_args.set_param_file_format("multiline")
+        show_iface_args.add(hs.tools.ghc.path)
+        show_iface_args.add("--show-iface ")
+        show_iface_args.add(hi_file.path)
+        hs.actions.write(show_iface_file, show_iface_args)
```

Comment:
@aherrmann, we have some controversy on what the interface of the GHC wrapper should be.

@GuillaumeGen is arguing that it is easier to accomplish the task if we keep this `show_iface_file`, while I think it is preferable to have the ghc_wrapper produce these arguments on its own for the sake of modularity (so the caller isn't concerned with how the callee performs the work).

Do you have any opinion on what the interface should be?

Reply:
> [..] I think it it is preferable to have the ghc_wrapper produce these arguments on its own for the sake of modularity (so the caller isn't concerned with how the callee performs the work).

Generally, I'd agree with this point of view.

> it is easier to accomplish the task if we keep this `show_iface_file`

However, I can see how this may be the case, given that `ghc_wrapper` is a bash script, and the information is coming in in a relatively unstructured format.

My impression is that `ghc_wrapper.sh` is growing beyond the scale where a bash script is still reasonable. A better approach may be the one also used by the Cabal actions: A `py_binary` that takes a JSON file as an input, such that the invoking rule can provide the needed metadata in a structured format. However, that's a wider refactoring. So, that would belong into its own PR.

Reply:
@GuillaumeGen, please, leave it as is for now.